app/grandchallenge/core/templates/grandchallenge/partials/drf_csrf.html

```diff
@@ -1,13 +1,7 @@
 {% load static %}
-{% load string %}
+{% load csrf %}
 
-{{ csrf_header_name|default:'X-CSRFToken'|json_script:"csrfHeaderName" }}
+{% csrf_data csrf_token as drf_data %}
```

Comment:
Don't you also need to pass `csrf_header_name`?

The `bool(request) == False` path doesn't seem to be handled?

Reply:
I made it so that you can optionally pass `csrf_header_name`. If you don't the default of "X-CSRFToken" is used. 

And why do we need to handle `bool(request) == False`? It will just be an empty string then right?

Reply:
> I made it so that you can optionally pass csrf_header_name. If you don't the default of "X-CSRFToken" is used.

Yes, but you're not sending `csrf_header_name` to the template tag function. It should be sent, right?

> And why do we need to handle bool(request) == False? It will just be an empty string then right?

Don't know. It was added here, I am not sure why https://github.com/comic/grand-challenge.org/pull/2891/commits/781e6c2195f87eb65c66b8bafca094289eff1e51. Maybe it was a problem with the `to_string` method. So okay, lets leave that out.

Reply:
> but you're not sending `csrf_header_name` to the template tag function. It should be sent, right?

No, if it's not sent along, its default value is used "X-CSRFToken".

Reply:
So why not send it along? In the original implementation if the value of `csrf_header_name` changes this bit of code automatically works. In this implementation if the value of `csrf_header_name` changes this bit of code will break again. As there is no automated test for it, it would only be caught in production.

Reply:
Yeah ok, that's a good point.